contracts/orbital-core/src/icq.rs

```diff
@@ -25,34 +25,9 @@ use neutron_sdk::interchain_queries::types::{
 };
 use serde_json_wasm;
 
-/// defines the incoming transfers limit to make a case of failed callback possible.
-const MAX_ALLOWED_TRANSFER: u64 = 20000;
-const MAX_ALLOWED_MESSAGES: usize = 20;
+use crate::state::{Transfer, RECIPIENT_TXS, TRANSFERS};
 
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-pub struct Cw20BalanceResponse {
-    pub balance: Uint128,
-}
-
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-pub struct GetRecipientTxsResponse {
-    pub transfers: Vec<Transfer>,
-}
-pub type Recipient = str;
-/// contains all transfers mapped by a recipient address observed by the contract.
-pub const RECIPIENT_TXS: Map<&Recipient, Vec<Transfer>> = Map::new("recipient_txs");
-/// contains number of transfers to addresses observed by the contract.
-pub const TRANSFERS: Item<u64> = Item::new("transfers");
-
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
-pub struct Transfer {
-    pub recipient: String,
-    pub sender: String,
-    pub denom: String,
-    pub amount: String,
-}
+const MAX_ALLOWED_MESSAGES: usize = 20;
```

Comment:
yeah, my bad. this is inspired from the neutron icq docs where `MAX_ALLOWED_MESSAGES` is a variable used for processing a message body. cosmos-sdk-proto `TxBody` contains the following field: `messages: ::prost::alloc::vec::Vec<::prost_types::Any>`. this can cause out of gas errors if we just blindly try to process the entire vec of messages.

so in our processing of that tx, we iterate over that tx body and take up to `MAX_ALLOWED_MESSAGES` from that iterator, ignoring anything beyond that range.